### DISCUSSION TITLE
Can't install web3.py through pip (unclear error)

### DISCUSSION THREAD

**Discussion by maindotmarcell** [smartcontractkit/full-blockchain-solidity-course-py#773] [Q&A] [answered]
Hey there,
"pip install web3" doesn't seem to work for me. I get the same error no matter what I do. I've tried reinstalling both python and pip, other packages install without any issues. I'm on Python 3.9.9 and pip 21.3.1.  I've even tried launching a virtual environment as directed in the web3.py documentation.
You can find some screenshots of the error message below:
![image](https://user-images.githubusercontent.com/71495475/149265865-cb579b5c-b80b-4e3e-8307-ee072a4338dc.png)
![image](https://user-images.githubusercontent.com/71495475/149265978-46c3a231-9f61-4589-8006-3bb00e9f6d07.png)

![image](https://user-images.githubusercontent.com/71495475/149266018-55629eb9-31c4-4da9-93db-517867a44869.png)
![image](https://user-images.githubusercontent.com/71495475/149266040-d9275e9c-2023-4bc7-b385-d8476571d69a.png)
![image](https://user-images.githubusercontent.com/71495475/149266076-dd828654-3e88-4b79-8d41-fe1819ac021f.png)

I have no idea what the issue is...
Thanks for any help in advance!

**Comment by maindotmarcell** [comment, marked as answer]
Solved it by updating my OS

**Reply by cromewar** [reply]
Nice, this was the cytools error a lot of users had, I'm glad an update helped you to get it done.

**Reply by mart-cervants** [reply]
I have the same error but I'm using Windows I'll try to install WSL, I hope this works
